### PR TITLE
enhancement: dont draw hittest when viewport animation is active

### PR DESCRIPTION
**Pull request by Keelaro1**

### CODE REVIEW

File: src/chart/model/hit-test-canvas.model.ts [hunk 1]
```diff
@@ -61,6 +63,14 @@ export class HitTestCanvasModel extends CanvasModel {
 		initCanvasWithConfig(this, chartConfig);
 		canvas.style.visibility = 'hidden';
 		this.enableUserControls();
+		this.canvasAnimation.animationIntervalIdSubject.subscribe(() => {
```
Comment (disyakidneyshot) [inline src/chart/model/hit-test-canvas.model.ts:66]: memory leak, no unsubsrcribe call

File: src/chart/animation/canvas-animation.ts [hunk 2]
```diff
@@ -19,6 +20,7 @@ export interface AnimationConfig {
 }
 
 export const VIEWPORT_ANIMATION_ID = 'VIEWPORT_ANIMATION';
+type AnimationsStatus = 'inProgress' | 'finished';
```
Comment (KirillBobkov) [inline src/chart/animation/canvas-animation.ts:23]: type AnimationsStatus = boolean

File: src/chart/animation/canvas-animation.ts [hunk 3]
```diff
@@ -34,6 +36,7 @@ export const VIEWPORT_ANIMATION_ID = 'VIEWPORT_ANIMATION';
 export class CanvasAnimation {
 	animationIntervalId?: number;
 	animations: StringTMap<Animation> = {};
+	animationsStatusSubject = new BehaviorSubject<AnimationsStatus>('finished');
```
Comment (KirillBobkov) [inline src/chart/animation/canvas-animation.ts:39]: animationInProgressSubject = new BehaviorSubject<AnimationsStatus>(false);

**Review by KirillBobkov — CHANGES_REQUESTED** [review]

**Comment by KirillBobkov** [comment]
Please, add new profiler screenshot after new changes will be applied

File: src/chart/model/scale.model.ts [hunk 4]
```diff
@@ -93,6 +95,12 @@ export class ScaleModel extends ViewportModel {
 				this.fireChanged();
 			}),
 		);
+		this.addRxSubscription(
+			this.canvasAnimation.animationsStatusSubject.subscribe(() => {
+				const animationsStatus = this.canvasAnimation.animationsStatusSubject.getValue();
+				this.hitTestCanvasModel.hitTestDrawersPredicateSubject.next(animationsStatus === 'finished');
+			}),
+		);
```
[the remark this answers is not in the document]
Comment (disyakidneyshot) [reply src/chart/model/scale.model.ts:103]: but scale model is not a nice place to insert a `hitTestCanvasModel`, keep in mind, that we should keep our entities as loose coupled as possible
What was your different ideas, where to put it?

**Review by disyakidneyshot — CHANGES_REQUESTED** [review]

**Review by disyakidneyshot — APPROVED** [review]

**Pull request merged**